### PR TITLE
nu-cli/completions: add test for variable completions

### PR DESCRIPTION
**Pull request by herlon214**
# Description

Add test for variable completions and refactor the completions tests files.

**Please check https://github.com/nushell/nushell/pull/5460 first.**

# Tests

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --all --all-features -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo build; cargo test --all --all-features` to check that all the tests pass

### CODE REVIEW

File: crates/nu-cli/tests/test_completions.rs [hunk 1]
```diff
@@ -116,8 +199,47 @@ pub fn new_engine() -> (PathBuf, String, EngineState) {
...
+
+        let (_, err) = parse(&mut working_set, None, record.as_bytes(), false, &[]);
+
+        assert!(err.is_none());
+        working_set.render()
+    };
+
+    // Merge delta
+    let merge_result = engine_state.merge_delta(delta, Some(&mut stack), &dir);
+    assert!(merge_result.is_ok());
```
Comment (herlon214) [inline crates/nu-cli/tests/test_completions.rs:240]: @kubouch am I doing this right?
In the test it still can't find the variable

File: crates/nu-cli/tests/support/completions_helpers.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+use nu_parser::parse;
+use nu_protocol::{
+    engine::{EngineState, Stack, StateDelta, StateWorkingSet},
+    PipelineData, Span, Value,
+};
+use nu_test_support::fs;
+use reedline::Suggestion;
+const SEP: char = std::path::MAIN_SEPARATOR;
+
+#[allow(dead_code)]
```
Comment (herlon214) [inline crates/nu-cli/tests/support/completions_helpers.rs:14]: If I remove those `#[allow(dead_code)]` compiler shows warning, even though it's being used.

**Comment by herlon214** [comment]
This PR is ready to be reviewed.

**Comment by fdncred** [comment]
> This PR is ready to be reviewed.

It looks like we have a conflict. Not sure if that just popped up or what.

**Pull request closed without merging**